qt/python/mantidqt/mantidqt/widgets/test/test_fitpropertybrowser.py

```diff
@@ -282,6 +283,24 @@ def test_fit_property_browser_correctly_updates_contents_from_fitted_func(self):
         # it should show a linear line with intercept 2 and gradient 1
         self.assertEqual(str(func.getFunction(0)), 'name=LinearBackground,A0=2,A1=1')
 
+    def test_fit_property_browser_correctly_handles_bin_plots(self):
+        # create & plot workspace
+        fig, canvas, ws = self._create_and_plot_matrix_workspace('ws_name', distribution=True)
+
+        #set bin plot kwargs
+        plot_kwargs = {"axis": MantidAxType.BIN}
+
+        #overplot bin plot
+        plot([ws], wksp_indices=[0], plot_kwargs=plot_kwargs, fig=fig, overplot=True)
+
+        property_browser = self._create_widget(canvas=canvas)
+        property_browser.show()
+
+        try:
+            property_browser.fit()
+        except:
+            self.fail("property_browser.fit failed to succesfully handle bin plot.")
```

Comment:
Could you put this in the form of an asserion that an error has been produced rather than a try catch?